test/crawlers/puppeteer_crawler.test.js

```diff
@@ -200,4 +200,38 @@ describe('PuppeteerCrawler', () => {
 
         expect(loadedUserAgent).toEqual(opts.userAgent);
     });
+
+    test('should set cookies assigned to session to page', async () => {
+        const cookies = [
+            {
+                name: 'example_cookie_name',
+                domain: '.example.com',
+                value: 'example_cookie_value',
+                expires: -1,
+            },
+        ];
+
+        let pageCookies;
+        let sessionCookies;
+
+        const puppeteerCrawler = new Apify.PuppeteerCrawler({
+            requestList,
+            useSessionPool: true,
```

Comment:
please add `persistCookiesPerSession: true` to make it clear. Defaults can change.

Reply:
Adding it - but as in the issue conversation - this should not really affect anything, as we specifically save cookies to session in `createSessionFunction` -> so it should not matter whether it's specified or not...

Reply:
It does not matter now. But it could matter a year from now 😉